std/fmt.zig

```diff
@@ -10,6 +10,22 @@ const lossyCast = std.math.lossyCast;
 
 pub const default_max_depth = 3;
 
+const FormatOptions = struct {
+    precision: ?usize = null,
+    width: ?usize = null,
+};
+
+fn nextArg(comptime used_pos_args: *u32, comptime maybe_pos_arg: ?comptime_int, comptime next_arg: *comptime_int) comptime_int {
```

Comment:
This was part of the motivation for #2710. I initially tried to handle this function in a function called `formatRaw` which performed this logic, and if successful, passed the correct value through to `formatType`. However, this required passing varargs, hence why it is now like this.